spec/automated_review/no_hardcoded_ids_spec.rb

```diff
@@ -0,0 +1,31 @@
+describe "No hardcoded ids" do
+  def hardcoded_ids_message(files)
+    <<~EOS
+      Hardcoded id numbers were found in the following locations:
+
+      #{files.join("\n").indent(4)}
+
+      Hardcoded id numbers should not be present in a data migration spec, as
+      they ignore region numbers, creating an invalid database setup. Instead,
+      use the following helper to generate ids in the correct region:
+
+          # Example, in region 1, requesting id number 2
+          anonymous_class_with_id_regions.id_in_region(2, anonymous_class_with_id_regions.my_region_number)
+          # => 1000000000002
+
+      If you believe this file is a false postive, add the following as a
+      tailing comment to the line in question
+
+          # manageiq:disable HardcodedIds
+
+    EOS
+  end
+
+  it "need tests" do
+    hardcoded_ids = Dir.chdir(ManageIQ::Schema::Engine.root) do
+      `grep -n "_\\?id\\s*=>\\?\\s*[0-9]" spec/**/*_spec.rb | grep -v "manageiq:disable HardcodedIds"`.strip.split("\n")
```

Comment:
Is that regex ignoring ruby 1.9 format: `Classification.where(parent_id: 0)`?  🤣

Reply:
yeah technically :)

Reply:
I tried adding in the 1.9 format, but I hit things like https://github.com/Fryguy/manageiq-schema/blob/f51b29941dc719c4e1870340151cd3f35fa7bc74/spec/migrations/20200212171537_remove_ovirt_sdk4_from_event_stream_spec.rb#L14, so I'm gonna punt for now

Reply:
yeah, handling one format will probably have false positives on it's own.  But yes, since 1.8 hash syntax is still most common, we can punt on 1.9 and even drop this for 1.8 if false positives happen that frequently.

Reply:
but wouldn't the ovirt example skip the grep because the ID has characters outside of 0-9?

Reply:
> but wouldn't the ovirt example skip the grep because the ID has characters outside of 0-9?

yeah, we'd have to ensure all values following the hashrocket are bigints.  I'm not sure if that will still leave us with false positives or not.

Reply:
> but wouldn't the ovirt example skip the grep because the ID has characters outside of 0-9?

If I spent enough time on it, yes :) I got up to `"_\?id\s*(=>\?|:)\s*[0-9]+\b"` but it still matched when it "starts with 0-9, but followed by a dash" - would need to specify "not immediately followed by a digit", but then you start getting into different grep implementations supporting negative lookahead or not, and I just couldn't be bothered anymore haha.